src/librbd/ImageCtx.h

```diff
@@ -250,7 +250,7 @@ namespace librbd {
 		  cls::rbd::SnapshotNamespace in_snap_namespace,
 		  librados::snap_t id,
 		  uint64_t in_size, parent_info parent,
-                  uint8_t protection_status, uint64_t flags);
+                  uint8_t protection_status, uint64_t flags, utime_t timestamp = utime_t());
```

Comment:
I found this default value will cause compilation failure of:
  MOCK_METHOD7(add_snap, void(std::string in_snap_name, librados::snap_t id,
                              uint64_t in_size, parent_info parent,
                              uint8_t protection_status, uint64_t flags, utime_t stamp));

Since I can not add default value for this stamp.

So I prefer delete the add_snap func call when create snap